### PR TITLE
Require `$schema` in ontology types

### PR DESCRIPTION
**Pull request by Alfred-Mountfield**
## 🌟 What is the purpose of this PR?

Unfortunately there was a typo where we updated the meta-schemas for the types to have the `schema` keyword instead of the `$schema` keyword. More importantly, there was an oversight and we didn't update the `type-system` definition to handle or enforce this.

This PR updates as necessary.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203358502199087/1204064995655460/f) _(internal)_

## 🚫 Blocked by

N/A

## 🔍 What does this change?

- Fixes the typos in the meta-schemas
- Updates the `type-system` package to enforce `$schema` to be set to the correct values
- Updates the API implementation to add `$schema` when creating types

## 📜 Does this require a change to the docs?

- The meta-schemas are part of the docs

## ⚠️ Known issues

- Any types currently deployed will need manually patching, we shall opt to do this while the published packages are still canaries. In the future this would need to be a proper update
- Our `ts-node` loader does _not_ like the `@blockprotocol/core` package. Importing it (even indirectly through `@blockprotocol/graph` causes strange errors complaining about CommonJS and named exports. it also causes problems within the site middleware:
  - [Patch broken seed-db](https://github.com/blockprotocol/blockprotocol/pull/1053/commits/3206ccbbf139c0c8bbe893300d2a639f239b28ad)
  - [Patch broken middleware](https://github.com/blockprotocol/blockprotocol/pull/1053/commits/629431153b0dd1ce4dd78cd149cff81619156c8e)
  - [Patch broken middleware again?](https://github.com/blockprotocol/blockprotocol/pull/1053/commits/0ec012af2770ee748d706b5fae124111862fa786)

## 🐾 Next steps

- Update the type editor
- Patch the DB
- Remove temporary patch on type-editor boundary

## 🛡 What tests cover this?

- Type System crate tests
- Type system node package tests

## ❓ How to test this?

1.  Checkout the branch / view the deployment
1.  `yarn` should pass, rebuilding the Type System crate
1.  `yarn workspace @blockprotocol/type-system test` should pass
1.  `cargo make test` should pass inside the `type-system/crate` directory 
1.  `cargo make lint` should pass inside the `type-system/crate` directory
1. Run the site and ensure the type-editor continues to function, and creates types with the correct fields

### CODE REVIEW

**Review by TimDiekmann — APPROVED** [review]
Rust changes look good to me. I'd probably prefer to have a static enum variant, but there is no point in blocking this PR on that. 

TS code also looks good, but probably should be reviewed from someone else as well.

File: apps/site/src/pages/api/types/entity-type/shared/schema.ts [hunk 1]
```diff
@@ -1,8 +1,9 @@
...
-  ENTITY_TYPE_META_SCHEMA,
-  EntityType,
-  EntityTypeWithMetadata,
-} from "@blockprotocol/graph";
+/*
+ @todo - We should be able to import from the `@blockprotocol/graph` package here but we're running into strange errors
+   with named exports, CommonJS, and ESM modules from the dependency on `/core`
+ */
+import { EntityType, EntityTypeWithMetadata } from "@blockprotocol/graph";
+import { ENTITY_TYPE_META_SCHEMA } from "@blockprotocol/type-system/slim";
```
Comment (Alfred-Mountfield) [inline apps/site/src/pages/api/types/entity-type/shared/schema.ts:6]: Doing this fixed the `seed-db` script which was causing problems with `ts-node` not understanding the `@blockprotocol/core` package

File: apps/site/src/middleware.page/return-types-as-json/hardcoded-types.ts [hunk 2]
```diff
@@ -1,7 +1,11 @@
+/*
+ @todo - We should be able to import from the `@blockprotocol/graph` package here but we're running into strange errors
+   with named exports, CommonJS, and ESM modules from the dependency on `/core`
+ */
 import {
   DATA_TYPE_META_SCHEMA,
   ENTITY_TYPE_META_SCHEMA,
-} from "@blockprotocol/graph";
+} from "@blockprotocol/type-system/slim";
```
Comment (Alfred-Mountfield) [inline apps/site/src/middleware.page/return-types-as-json/hardcoded-types.ts]: Doing this fixed _local_ site building which wasn't understanding the `@blockprotocol/core` package and erroring with:

```
Dynamic Code Evaluation (e. g. 'eval', 'new Function', 'WebAssembly.compile') not allowed in Edge Runtime 
Learn More: https://nextjs.org/docs/messages/edge-dynamic-code-evaluation
Import trace for requested module:
../../node_modules/es-module-lexer/dist/lexer.js
../../libs/@blockprotocol/core/dist/esm/html.js
../../libs/@blockprotocol/core/dist/esm/index.js
../../libs/@blockprotocol/graph/dist/es/main-non-temporal.js
./src/middleware.page/return-types-as-json/hardcoded-types.ts
./src/middleware.page/return-types-as-json.ts
```

File: apps/site/src/middleware.page/return-types-as-json/hardcoded-types.ts [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
+   the integration tests CI job
  */
-import {
-  DATA_TYPE_META_SCHEMA,
-  ENTITY_TYPE_META_SCHEMA,
-} from "@blockprotocol/type-system/slim";
+const DATA_TYPE_META_SCHEMA =
+  "https://blockprotocol.org/types/modules/graph/0.3/schema/data-type";
+const ENTITY_TYPE_META_SCHEMA =
+  "https://blockprotocol.org/types/modules/graph/0.3/schema/entity-type";
```
Comment (Alfred-Mountfield) [inline apps/site/src/middleware.page/return-types-as-json/hardcoded-types.ts:9]: Even though the [previous commit fixed site building both locally and on Vercel](https://github.com/blockprotocol/blockprotocol/pull/1053/commits/629431153b0dd1ce4dd78cd149cff81619156c8e#r1118761439), [for some reason the _Site integration tests_ job couldn't build it](https://github.com/blockprotocol/blockprotocol/actions/runs/4282761095/jobs/7457881561#step:8:36)?!

```
Failed to compile.

../../libs/@blockprotocol/type-system/dist/es-slim/index-slim.js
Dynamic Code Evaluation (e. g. 'eval', 'new Function', 'WebAssembly.compile') not allowed in Edge Runtime 
Learn More: https://nextjs.org/docs/messages/edge-dynamic-code-evaluation

Import trace for requested module:
../../libs/@blockprotocol/type-system/dist/es-slim/index-slim.js
./src/middleware.page/return-types-as-json/hardcoded-types.ts
./src/middleware.page/return-types-as-json.ts
```

File: libs/@blockprotocol/graph/src/shared/types/ontology/property-type.ts [hunk 4]
```diff
@@ -25,7 +25,7 @@ export type GetPropertyTypeData = {
   propertyTypeId: VersionedUrl;
 };
 
-type SystemDefinedPropertyTypeProperties = "$id" | "kind";
+type SystemDefinedPropertyTypeProperties = "schema" | "$id" | "kind";
```
Comment (thehabbos007) [inline libs/@blockprotocol/graph/src/shared/types/ontology/property-type.ts:28]:
```suggestion
type SystemDefinedPropertyTypeProperties = "$schema" | "$id" | "kind";
```

**Pull request merged**